ggshield/scan/scan_context.py

```diff
@@ -1,12 +1,56 @@
+import platform
+import sys
 import uuid
 from dataclasses import dataclass
-from typing import Dict, Optional, Union
+from functools import lru_cache
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
 
 from ggshield import __version__
 
 from .scan_mode import ScanMode
 
 
+@lru_cache(None)
+def get_os_info() -> Tuple[str, str]:
+    if sys.platform.lower() == "linux":
+        return parse_os_release(Path("/etc/os-release"))
+    else:
+        return platform.system().lower(), platform.version()
+
+
+def parse_os_release(os_release_path: Path) -> Tuple[str, str]:
+    """
+    Extract and return Linux's OS-name and OS-Version
+    If extraction fails, we return ('linux', 'unknown')
+    """
+    error_tuple = "linux", "unknown"
+
+    try:
+        with open(os_release_path) as f:
+            lines = f.readlines()
+    except FileNotFoundError:
+        return error_tuple
+    except PermissionError:
+        return error_tuple
+
+    # Build a dictionary from the os-release file contents
+    data_dict = {}
+    for line in lines:
+        if "=" in line:
+            key, value = line.split("=")
+            key, value = key.strip(), value.strip()
+            data_dict[key] = value.strip('"')
+
+    if "ID" not in data_dict:
+        return error_tuple
+
+    if "VERSION_ID" not in data_dict:
+        data_dict["VERSION_ID"] = "unknown"
+
+    return data_dict["ID"], data_dict["VERSION_ID"]
```

Comment:
```suggestion
    return data_dict["ID"], data_dict.get("VERSION_ID", "unknown")
```

Reply:
While catching individual exceptions is often the right thing to do, in this case I'd rather have a large try..except block. Something like this:

```python
try:
   # All the code, without exception handling
except Exception as exc:
   logger.warning("Failed to read Linux OS name and version: %s", exc)
   return error_tuple
```

This is because we are in a similar situation as `ggshield/core/check_updates.py`: this is code which is useful to us, but should never stop the user if it fails.

Reply:
At first, I thought it'd be better to catch the exceptions individually so that we can handle them gracefully one day _maybe_. But yes I agree, I'll bundle it all in one large try/catch.